modules/core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/BulkAdminServiceImpl.java

```diff
@@ -47,6 +47,13 @@
 
     public static final String BULK_SCROLL_PRODUCE_IMMEDIATE_PROPERTY = "nuxeo.core.bulk.scroller.produceImmediate";
 
+    // @since 11.4
+    public static final String BULK_SCROLL_PRODUCE_IMMEDIATE_THRESHOLD_PROPERTY = "nuxeo.core.bulk.scroller.produceImmediateThreshold";
+
+    // by default switch to produce immediate when there are more than 1m ids
+    // @since 11.4
+    public static final int DEFAULT_PRODUCE_IMMEDIATE_THRESHOLD_PROPERTY = 1_000_000;
```

Comment:
This constant isn't used...

Reply:
yes thanks, the pr has been reworked a bit to fix this and to flush accumulated records downstream when reaching the threshold avoiding a reordering of ids.